File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.py
def cal(k):
    if k == 0:
        return 1
    elif k < 0:
        return 0
    
    return cal(k - 1) + cal(k - 2) + cal(k - 3)


n = int(input()) 

l = []

for i in range(0,n):
    l.append(int(input()))

for k in l:
    print(cal(k))

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.cc
#include<bits/stdc++.h>
using namespace std;

int a, n;

int cal(int n){
	if(n==0) return 1;
	else if(n<0) return 0;
	
	return cal(n-1) + cal(n-2) + cal(n-3);
}

int main(){

	cin >> n;
	
	for(int i=0;i<n;i++){
		cin >> a;
		cout << cal(a) << "\n";
		
	}
	
	return 0;
}